File: Java/6 kyu Highest Scoring Word/Kata.java
/* 
Given a string of words, you need to find the highest scoring word.

Each letter of a word scores points according to its position in the alphabet: a = 1, b = 2, c = 3 etc.

For example, the score of abad is 8 (1 + 2 + 1 + 4).

You need to return the highest scoring word as a string.

If two words score the same, return the word that appears earliest in the original string.

All letters will be lowercase and all inputs will be valid.
*/

// My solution

public class Kata {

    public static String high(String s) {
        String[] words = s.split(" ");
        int maxScore = 0;
        String highestWord = "";

        for (String word : words) {
            int score = 0;

            for (char c : word.toCharArray()) {
                score += c - 'a' + 1;
            }

            if (score > maxScore) {
                maxScore = score;
                highestWord = word;
            }
        }

        return highestWord;
    }

}

File: Go/8 kyu Beginner Lost Without a Map/Kata.go
/*
Given an array of integers, return a new array with each value doubled.

For example:

[1, 2, 3] --> [2, 4, 6]
*/

package kata

func Maps(x []int) []int {
	result := make([]int, len(x))
	for i, v := range x {
		result[i] = v * 2
	}
	return result
}
